From 7604481131a9be2b93e13b1f1ed23ffb10e9084e Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Wed, 18 Jun 2014 08:27:33 +0200
Subject: [PATCH 2/2] Don't reset background when starting the X server

This way we won't loose the Plymouth splash screen.
See issue #196.
---
 src/daemon/DisplayServer.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/daemon/DisplayServer.cpp b/src/daemon/DisplayServer.cpp
index e520740..9ba8f85 100644
--- a/src/daemon/DisplayServer.cpp
+++ b/src/daemon/DisplayServer.cpp
@@ -71,7 +71,14 @@ namespace SDDM {
             process->setProcessEnvironment(env);
 
             // start display server
-            process->start(daemonApp->configuration()->serverPath(), { m_display, "-auth", m_authPath, "-nolisten", "tcp", QString("vt%1").arg(m_displayPtr->terminalId()) });
+            QStringList args;
+            args << m_display
+                 << "-auth" << m_authPath
+                 << "-nolisten" << "tcp"
+                 << "-background" << "none"
+                 << "-noreset"
+                 << QString("vt%1").arg(m_displayPtr->terminalId()),
+            process->start(daemonApp->configuration()->serverPath(), args);
         }
 
         // wait for display server to start
-- 
2.0.0

